src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHelper.Digest.cs

```diff
@@ -55,7 +58,15 @@ internal partial class AuthenticationHelper
             if (string.IsNullOrEmpty(parameter))
                 return false;
 
-            request.Headers.Authorization = new AuthenticationHeaderValue(Digest, parameter);
+            if (authHeader == HttpKnownHeaderNames.Authorization)
```

Comment:
Rather than passing in the authHeader name, why not just return the AuthenticationHeaderValue and let the caller set whichever header is appropriate?